src/Snap/Internal/Http/Server/Socket.hs

```
@@ -59,6 +63,29 @@ bindSocketImpl _setSocketOption _bindSocket _listen bindAddr bindPort = do
         _listen sock 150
         return $! sock
 
+bindUnixSocket :: Maybe Int -> String -> IO Socket
+#if HAS_SENDFILE
+bindUnixSocket perms path = do
+   when (not isAbsPath) $
+      throwIO (AddressNotSupportedException path)
+   bracketOnError (socket N.AF_UNIX N.Stream 0) N.close $ \sock -> do
+      exists <- fileExist path
```

This will also eat exception thrown if user has insufficient rights